plugins/async-node/jvm/src/test/kotlin/com/intuit/playerui/plugins/asyncnode/AsyncNodePluginTest.kt

```diff
@@ -228,4 +232,162 @@ internal class AsyncNodePluginTest : PlayerTest() {
         }
         Assertions.assertTrue(true)
     }
+
+    @TestTemplate
+    fun `handle multiple updates through callback mechanism`() = runBlockingTest {
+        // TODO: This typing is not great - need to go fix hook types
+        var deferredResolve: ((asyncNodeUpdate) -> Unit)? = null
+        var updateContent: ((asyncNodeUpdate) -> Unit)? = null
+
+        var count = 0
+
+        plugin.hooks.onAsyncNode.tap("") { _, node, callback ->
+            updateContent = callback
+            val result = suspendCoroutine { cont ->
+                deferredResolve = { value ->
+                    cont.resume(value)
+                }
+            }
+            BailResult.Bail(result)
+        }
+
+        var viewUpdateContinuation: Continuation<Asset?>? = null
```

Comment:
hmm is this assigned anywhere? what's the thinking behind this, I only see this being assigned as null

Reply:
This is used [here](https://github.com/player-ui/player/pull/488/files/691749196a68e7fedfade8be8ac931f085c95698#diff-ac6259632cb11fb85e6116a391926ea88917bc5f23faba6c0657f014a09d74c0R258) When an update is received, the continuation is resumed with the new asset, allowing the coroutine to continue.

Reply:
that linked spot, is it not `null` at that point? I'm curious because I can't seem to find a place where this is assigned to something other than null

Reply:
removed , its just for testing